peers.py

```diff
@@ -0,0 +1,58 @@
+from charms.reactive import RelationBase
+from charms.reactive import hook
+from charms.reactive import scopes
+from charms.reactive import not_unless
+
+
+class EtcdPeer(RelationBase):
+    ''' ETCD Peering works in tandem with the leader node to establish a cluster
+        as sanely as possible. This works statically behind a firewall without
+        any need for an external service to function as the discovery mechanism
+
+        Every peer participating in the cluster declares themselves to the
+        cluster. The leader then ingests the data and leader-set's the cluster
+        state (string, status, token) for the peers to ingest and participate.
+    '''
+    scope = scopes.UNIT
+
+    auto_accessors = ['public_address', 'unit_name', 'port']
```

Comment:
You might want to leave a comment about auto_accessors for people like "Kevin" who don't understand exactly what they do and why they are cool.